NicolaeChristof/Rise
# Feature levelselect, endgame, and gameover

There are three major things that I implemented in this branch that need to be tested.

1. Level Select

Description: 
Level Select now appears in the main menu and selecting that option will take you to another menu that has the four template levels currently in the game. Selecting one of those will take you to that template scene. Since all this was done in the canvas prefab the other scenes have their own duplicate menu that you will need to go through to start but everything else should be in order. I assume the duplicate canvas stuff will change after we discuss more about UI management on Monday.

How to Test:  Just select the level select option and test if it takes you to the correct season. You can go from one season menu to another with level select currently due to how the canvas prefab works.


2. Endgame Menu

Description:
Reaching the top now opens an endgame menu that lets you return to main menu or go to the next level. Currently the levels cycle from spring->summer->fall->winter->spring but that will change as we add more levels. 

How to Test: 
Starting the game in single-player puts you in the test scene with a teleporter directly to the top (F on the keyboard to use). Hitting Next Level should take you to the spring scene which looks similar but there won't be a teleporter in the scene, You can then reach the top from there and it will go to summer, so on and so forth.


3. Game-over Menu

Description:
The game now has a game-over state when you run out of health, where it normally paused the game before and had no menu. You can now restart the level or return to main menu from here.

How to Test:
Just die and then test to see if the menu works.
